h2o-core/src/test/java/water/TestUtil.java

```diff
@@ -151,6 +151,23 @@ public static void checkLeakedKeys() {
     _initial_keycnt = H2O.store_size();
   }
 
+  public static void checkArrays(double[] expected, double[] actual, double threshold) {
+    for(int i = 0; i < actual.length; i++) {
+      if (!Double.isNaN(expected[i]) && !Double.isNaN(actual[i])) // only compare when both are not NaN
+        assertEquals(expected[i], actual[i], threshold*Math.max(Math.abs(expected[i]),Math.abs(actual[i])));
```

Comment:
I don't understand `threshold*Math.max(Math.abs(expected[i]),Math.abs(actual[i]))`

Do you intend to make the actual resulting threshold to be a fraction of the greater of the two compared numbers ? Couldn't this result in big inaccuracies ?

Reply:
Pavel:  

Yes, you are correct in the absolute sense.  However, the difference caused by numerical instability scaled with the magnitude of the numbers.  In order to get rid of the effect of this magnitude scaling, I set the tolerance to scale with the number magnitude. 

However, I think I should use the Math.min instead of Math.max in this case.